### PR TITLE
implemented search state changes notification

### PR DESCRIPTION
**Pull request by DFSko**

### CODE REVIEW

File: Api/Services/Hubs/Search/SearchHub.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        }
+        
+        
+        public override async Task OnConnectedAsync()
+        {
+            var identityId = Context.User?.FindFirstValue("sub");
+            if (string.IsNullOrEmpty(identityId))
+                return;
+
+            var agent = await _context.Agents.FirstOrDefaultAsync(a => a.IdentityHash == HashGenerator.ComputeSha256(identityId));
```
Comment (mingazhev) [inline Api/Services/Hubs/Search/SearchHub.cs:27]: SingleOrDefaultAsync

File: Api/Models/Availabilities/SearchStateChangedToastMessage.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+using System;
+
+namespace HappyTravel.Edo.Api.Models.Availabilities
+{
+    public readonly struct SearchStateChangedToastMessage
```
Comment (mingazhev) [inline Api/Models/Availabilities/SearchStateChangedToastMessage.cs:5]: This is not a toast message, maybe just a message?

File: Api/Services/Accommodations/Availability/Steps/WideAvailabilitySearch/WideAvailabilitySearchTask.cs [hunk 3]
```diff
@@ -73,6 +78,7 @@ await GetAvailability(connectorRequest, languageCode)
                     .Map(ApplyMarkups)
                     .Map(Convert)
                     .Tap(SaveResult)
+                    .Tap(SendToastMessage)
```
Comment (mingazhev) [inline Api/Services/Accommodations/Availability/Steps/WideAvailabilitySearch/WideAvailabilitySearchTask.cs:81]: NotifyClient maybe? this is not a toast message

File: Api/Services/Hubs/Search/SearchHub.cs (hunk 1 above)
Comment (DFSko) [reply Api/Services/Hubs/Search/SearchHub.cs]: Fixed

File: Api/Models/Availabilities/SearchStateChangedToastMessage.cs (hunk 2 above)
Comment (DFSko) [reply Api/Models/Availabilities/SearchStateChangedToastMessage.cs]: Fixed

File: Api/Services/Accommodations/Availability/Steps/WideAvailabilitySearch/WideAvailabilitySearchTask.cs (hunk 3 above)
Comment (DFSko) [reply Api/Services/Accommodations/Availability/Steps/WideAvailabilitySearch/WideAvailabilitySearchTask.cs]: Fixed

**Review by mingazhev — APPROVED** [review]

**Review by AlexeyShok — APPROVED** [review]

**Pull request merged**